packages/flutter/lib/src/rendering/box.dart

```diff
@@ -583,12 +585,15 @@ class BoxConstraints extends Constraints {
       return true;
     if (runtimeType != other.runtimeType)
       return false;
-    final BoxConstraints typedOther = other;
-    assert(typedOther.debugAssertIsValid());
-    return minWidth == typedOther.minWidth &&
-           maxWidth == typedOther.maxWidth &&
-           minHeight == typedOther.minHeight &&
-           maxHeight == typedOther.maxHeight;
+    assert(() {
```

Comment:
Can this not just be
`assert(other is BoxConstraints && other.debugAssertIsValid());`?

Reply:
Good catch